File: services/auth/src/main/java/com/example/iter/auth/service/RefreshTokenHasher.java
package com.example.iter.auth.service;

import org.springframework.stereotype.Component;
import org.springframework.util.StringUtils;

import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.Base64;

@Component
public class RefreshTokenHasher {

    public String hash(String refreshToken) {
        if (!StringUtils.hasText(refreshToken)) {
            throw new IllegalArgumentException("Refresh Token은 필수입니다.");
        }

        try {
            MessageDigest digest = MessageDigest.getInstance("SHA-256");
            byte[] hashed = digest.digest(refreshToken.getBytes(StandardCharsets.UTF_8));
            return Base64.getUrlEncoder().withoutPadding().encodeToString(hashed);
        } catch (NoSuchAlgorithmException e) {
            throw new IllegalStateException("SHA-256 해시 알고리즘을 사용할 수 없습니다.", e);
        }
    }
}

File: services/auth/src/main/kotlin/com/example/iter/auth/service/RefreshTokenHasher.kt
package com.example.iter.auth.service

import org.springframework.stereotype.Component
import org.springframework.util.StringUtils
import java.nio.charset.StandardCharsets
import java.security.MessageDigest
import java.security.NoSuchAlgorithmException
import java.util.Base64

@Component
class RefreshTokenHasher {

    fun hash(refreshToken: String): String {
        if (!StringUtils.hasText(refreshToken)) {
            throw IllegalArgumentException("Refresh Token은 필수입니다.")
        }

        try {
            val digest = MessageDigest.getInstance("SHA-256")
            val hashed = digest.digest(refreshToken.toByteArray(StandardCharsets.UTF_8))
            return Base64.getUrlEncoder().withoutPadding().encodeToString(hashed)
        } catch (e: NoSuchAlgorithmException) {
            throw IllegalStateException("SHA-256 해시 알고리즘을 사용할 수 없습니다.", e)
        }
    }
}
